tests/ottest/fixtures/blockchain/Restart.cpp

```diff
@@ -0,0 +1,264 @@
+    EXPECT_EQ(
+        GetBalance(user_bob),
+        amount_in_transaction_ * blocks_number_ * transaction_in_block_);
+    EXPECT_EQ(GetTransactions(user_bob).size(), 1);
+    EXPECT_EQ(GetTransactions(user_alice).size(), 1);
+}
+
+void Restart_fixture::mineTransaction(
+    const User& user,
+    const opentxs::blockchain::block::pTxid& transactionToConfirm)
+{
+    std::vector<Transaction> vec;
+
+    auto send_transaction =
+        user.api_->Crypto().Blockchain().LoadTransactionBitcoin(
+            transactionToConfirm);
+    vec.emplace_back(std::move(send_transaction));
+
+    Mine(targetHeight++, vec.size(), default_, vec);
+}
+
+void Restart_fixture::sendCoins(const User& receiver, const User& sender)
+{
+    const auto& network =
+        sender.api_->Network().Blockchain().GetChain(test_chain_);
+
+    const ot::UnallocatedCString memo_outgoing = "test";
+
+    const auto address = GetNextBlockchainAddress(receiver);
+
+    std::cout << sender.name_ + " send " << coin_to_send_ << " to "
+              << receiver.name_ << " address: " << address << std::endl;
+
+    auto future = network.SendToAddress(
+        sender.nym_id_, address, coin_to_send_, memo_outgoing);
+
+    // We need to confirm send transaction, so it is available after restore
+    mineTransaction(sender, future.get().second);
+}
+
+void Restart_fixture::collectOutputs(
+    const User& user_bob,
+    const User& user_alice,
+    Vector<UTXO>& all_bobs_outputs,
+    Vector<UTXO>& all_alice_outputs,
+    std::map<TxoState, std::size_t>& number_of_outputs_per_type_bob,
+    std::map<TxoState, std::size_t>& number_of_outputs_per_type_alice)
+{
+    const auto& bobs_network =
+        user_bob.api_->Network().Blockchain().GetChain(test_chain_);
+    const auto& bobs_wallet = bobs_network.Wallet();
+
+    const auto& alice_network =
+        user_alice.api_->Network().Blockchain().GetChain(test_chain_);
+    const auto& alice_wallet = alice_network.Wallet();
+
+    all_bobs_outputs = bobs_wallet.GetOutputs(TxoState::All);
+    all_alice_outputs = alice_wallet.GetOutputs(TxoState::All);
+
+    auto all_types = {
+        TxoState::Immature,
+        TxoState::ConfirmedSpend,
+        TxoState::UnconfirmedSpend,
+        TxoState::ConfirmedNew,
+        TxoState::UnconfirmedNew,
+        TxoState::OrphanedSpend,
+        TxoState::OrphanedNew};
+    for (const auto& type : all_types) {
+        number_of_outputs_per_type_bob[type] =
+            bobs_wallet.GetOutputs(type).size();
+        number_of_outputs_per_type_alice[type] =
+            alice_wallet.GetOutputs(type).size();
+    }
+}
+
+void Restart_fixture::collectOutputsAsSet(
+    const User& user_bob,
+    const User& user_alice,
+    OutputsSet& all_bobs_outputs,
+    OutputsSet& all_alice_outputs,
+    std::map<TxoState, std::size_t>& number_of_outputs_per_type_bob,
+    std::map<TxoState, std::size_t>& number_of_outputs_per_type_alice)
+{
+    // Wait for outputs to process
+    sleep(20);
```

Comment:
worked for us in the test, but might not be the required default behavior if anyone uses this method elsewhere ;)